backend/src/test/java/de/armbrust/planz/service/ProductServiceTest.java

```diff
@@ -0,0 +1,97 @@
+package de.armbrust.planz.service;
+
+import de.armbrust.planz.db.ProductMongoDb;
+import de.armbrust.planz.model.Product;
+import org.hamcrest.Matchers;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+
+import java.util.List;
+import java.util.Optional;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.containsInAnyOrder;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static org.mockito.Mockito.*;
+
+class ProductServiceTest {
+
+    private final ProductMongoDb productMongoDb = mock(ProductMongoDb.class);
+    private final ProductService productService = new ProductService(productMongoDb);
+
+    @Test
+    @DisplayName("List Products should return list from DB")
+    public void listProducts() {
+        //GIVEN
+        when(productMongoDb.findAll()).thenReturn(List.of(
+                Product.builder()
+                        .sku("123")
+                        .title("Product1")
+                        .asin("ABC")
+                        .build(),
+                Product.builder()
+                        .sku("456")
+                        .title("Product2")
+                        .asin("EFG")
+                        .build())
+        );
+        //WHEN
+        List<Product> products = productService.listProducts();
+
+        //THEN
+        assertThat(products, containsInAnyOrder(
+                Product.builder()
+                        .sku("123")
+                        .title("Product1")
+                        .asin("ABC")
+                        .build(),
+                Product.builder()
+                        .sku("456")
+                        .title("Product2")
+                        .asin("EFG")
+                        .build()));
+    }
+
+
+    @Test
+    @DisplayName("A not existing product should be added to db")
+    public void testAddNotExistingProduct() {
+        //GIVEN
+        Optional<Product> OptionalOfProductToAdd = Optional.of(Product.builder()
```

Comment:
use lowercase letters for variable names. Moreover, I would remove the "OptionalOf" since it's obvious through the declaration.

Not sure if you need the Optional here at all. You could simply pass a product to the addProduct method in line 69, instead of an optional.
```suggestion
        Product productToAdd = Product.builder()
```

Reply:
I get a Optional<Product> return from 'productService.addProduct'. I used Optional<Product> in line 60 to compare with the return.

Reply:
Got it! You could then also rewrite line 74 to assertThat(actual.get(), Matchers.is(productToAdd)) but that's just personal preference I guess :)